src/seedu/addressbook/Main.java

```diff
@@ -24,7 +25,7 @@
 
     /** Version info of the program. */
     public static final String VERSION = "AddressBook Level 2 - Version 1.0";
-
+    public static Stack<String> CommandHistory = new Stack<>();
```

Comment:
Does this need to be public?

Reply:
I have changed it to private in Main.java now, however it is public in Parser.java and HistoryCommand.java. I think that is needed because I will be passing the CommandHistory Stack from main to parser to the the command handling class. Please let me know if this can be done in a better way and why this approach is not preferred.

Reply:
It isn't that its not preferred, as long as you have a reason to make things publicly accessible then that's fine :)